### PR TITLE
Add device query variable for selecting device

### PR DESCRIPTION
**Pull request by glistening**
JSRemoteTest-DCO-1.0-Signed-off-by: Sanggyu Lee sg5.lee@samsung.com

### CODE REVIEW

**Comment by glistening** [comment]
The code `get_parameter_by_name` is copied from http://stackoverflow.com/questions/901115/get-querystring-values-with-jquery/5158301#5158301. I just changed the camel style to snake style.

**Comment by rtakacs** [comment]
I think the webpage could seem buggy for an average user, if no device selected (empty measurement charts with Loading... label, but nothing happens).

Instead, js-remote-test could has a default webpage with a little use-case information and also could show the summary of the latest testresult for both targets (without charts):

```
     Welcome to js-remote-test

  The js-remote-test webpage is for ...
        ...
        ...

  artik053
      105 pass 0 fail 26 skip

  stm32f4-discovery
      82 pass 0 fail 49 skip

  rpi2
      99 pass 0 fail 32 skip

  Select the concrete target for more information.

```

**Comment by robertsipka** [comment]
I agree with @rtakacs or at least it should show a message that there is no device selected.

**Comment by yichoi** [comment]
LGTM

**Comment by glistening** [comment]
@robertsipka @rtakacs Thank you for comments. What about setting artik053 as default target as it was? We can make issue and discuss and make another PR based on conclusion.

**Comment by rtakacs** [comment]
@glistening Yes, we'll create a home page for js-remote-test (if no device is selected). After that, this PR could be landed. ;)

**Comment by glistening** [comment]
@rtakacs Okay. For your information, I updated this PR. When no device is specified, it shows artik053 information. It is same to the previous behavior.

File: js/load_jsons.js [hunk 1]
```diff
@@ -3,7 +3,8 @@ var g_db_keys = [];
 var g_db_ref = "";
 
 $(document).ready(function() {
-  change_device("artik053");
+  var device = get_parameter_by_name('device') || 'artik053';
+  change_device(device);
```
Comment (rtakacs) [inline js/load_jsons.js]: Maybe the parameter should be validated. 

```
supported_devices = ['stm32', 'rpi2', 'artik053'];

// Set artik053 as a default device
if (!(supported_devices.includes(device)))
  device = 'artik053';
```

File: js/load_jsons.js [hunk 2]
```diff
@@ -84,3 +85,8 @@ function fetch_keys(device) {
 
     });
 }
+
+function get_parameter_by_name(name) {
+    var match = RegExp('[?&]' + name + '=([^&]*)').exec(window.location.search);
+    return match && decodeURIComponent(match[1].replace(/\+/g, ' '));
```
Comment (rtakacs) [inline js/load_jsons.js]: It's enough only 2 spaces for indentation.

Comment (glistening) [reply js/load_jsons.js]: @rtakacs I will fix it.

File: js/load_jsons.js (hunk 1 above)
Comment (glistening) [reply js/load_jsons.js]: @rtakacs I don't think it is natural to redirect to `artik053` for every requests that is not in supported devices. And I belive most users don't type the url query string in url bar directly. They will just click, and it is not difficult to identify the typo or bad selection of device. But I will update as you commented. I don't want to put much time on it.

Comment (rtakacs) [reply js/load_jsons.js]: @glistening You are correct. It is a bit strange to add artik053 as a default device if the parameter is stm32f4 (and not the required stm32).
In the next Pull Request, the default case (if no device selected, or the device parameter is not correct) will be the default web-page, so artik053 will be eliminated as a default device.

**Comment by robertsipka** [comment]
LGTM

File: js/load_jsons.js (hunk 1 above)
Comment (glistening) [reply js/load_jsons.js]: @rtakacs Okay. I selected `artik053` as default page to keep the behaviors same to that before this PR.

**Pull request merged**